radixdlt-java/src/main/java/com/radixdlt/client/application/RadixApplicationAPI.java

```diff
@@ -128,12 +128,39 @@ public Completable toCompletable() {
 
 	private final Ledger ledger;
 
+	private RadixApplicationAPI(
+		RadixIdentity identity,
+		RadixUniverse universe,
+		FeeMapper feeMapper,
+		Ledger ledger,
+		List<ActionToParticlesMapper> actionToParticlesMappers
+	) {
+		this.identity = identity;
+		this.universe = universe;
+
+		// TODO: Utilize class loader to discover and load these modules
+		this.uniquePropertyTranslator = new UniquePropertyTranslator();
+		this.messageActionStore = new ActionStore<>(ledger.getAtomStore(), new AtomToDecryptedMessageMapper(universe));
+		this.tokenBalanceStore = new ApplicationStore<>(ledger.getParticleStore(), new TokenBalanceReducer());
+		this.tokenTransferActionStore = new ActionStore<>(ledger.getAtomStore(), new AtomToTokenTransfersMapper(universe));
+		this.tokenStore = new ApplicationStore<>(ledger.getParticleStore(), new TokenReducer());
+
+		this.actionToParticlesMappers = actionToParticlesMappers;
+		this.feeMapper = feeMapper;
+		this.ledger = ledger;
```

Comment:
Maybe null-check these too? Especially the `actionToParticlesMapper` (which, actually, could be tested for emptiness as well) as it ultimately comes from the caller.

Reply:
Yes! thanks for catching that.